cartography/intel/crxcavator/crxcavator.py

```diff
@@ -0,0 +1,220 @@
+    :return: Returns JSON text blob for the API called. API spec is at https://crxcavator.io/apidocs
+    """
+    uri = crxcavator_base_url + api_and_parameters
+    data = requests.get(
+        uri,
+        headers={'Accept': 'application/json',
+                 'API-Key': crxcavator_api_key})
+    # if call failed, use requests library to raise an exception
+    data.raise_for_status()
+    return data.json()
+
+
+def transform_extensions(extension_json):
+    """
+    Transforms the raw extensions JSON from the API into a list of extensions data
+    :param extension_json:  The JSON text blob returned from the CRXcavator API
+    :return: List containing extension info for ingestion
+    """
+    # the JSON returned from the CRXcavator API does not return well formatted objects
+    # instead, each object is named after it's key, making enumeration more difficult
+    # will build a cleaner object for import into graph
+
+    _data_index = 1
+    _row_index = 0
+
+    extensions = []
+    for extension in extension_json.items():
+        details = extension[_data_index][_row_index]
+        if not details:
+            continue
+        extension_id = details['extension_id']
+        version = details['version']
+        data = details['data']
+        # permissions warnings can be a list or null, if list concat to a single string
+        permission_warnings = data['webstore'].get('permission_warnings')
+        if permission_warnings:
+            permission_warnings = ', '.join(permission_warnings)
+        extensions.append({
+            'id': "{0}|{1}".format(extension_id, version),
+            'extension_id': extension_id,
+            'version': version,
+            'risk_total': data['risk'].get('total'),
+            'risk_metadata': json.dumps(data['risk'].get('metadata')),
+            'address': data['webstore'].get('address'),
+            'email': data['webstore'].get('email'),
+            'icon': data['webstore'].get('icon'),
+            'last_updated': data['webstore'].get('last_updated'),
+            'name': data['webstore'].get('name'),
+            'offered_by': data['webstore'].get('offered_by'),
+            'permissions_warnings': permission_warnings,
+            'privacy_policy': data['webstore'].get('privacy_policy'),
+            'rating': data['webstore'].get('rating'),
+            'rating_users': data['webstore'].get('rating_users'),
+            'short_description': data['webstore'].get('short_description'),
+            'size': data['webstore'].get('size'),
+            'support_site': data['webstore'].get('support_site'),
+            'users': data['webstore'].get('users'),
+            'website': data['webstore'].get('website'),
+            'type': data['webstore'].get('type'),
+            'price': data['webstore'].get('price')
+        })
+    if len(extensions) == 0:
+        raise ValueError('No extensions could be parsed from JSON data')
```

Comment:
Is there cases where we could have no extensions? For example, if no extensions are installed by users, would we hit this or never get here? I'm worried about throwing exception here and crashing sync when we might just log.

Reply:
Prior version had log and return, but discussion with ecdavis was that preferred method was to fail sync on unrecoverable error. While it is possible no extensions would be returned if none are installed, if they have configured this module this case should not occur and an error would be warranted.

Reply:
What about day 1 setup where they got the service going but waiting for the extension to deploy? Might be edge case but still valid to avoid hard crash. Not PR blocker.

Reply:
If there are no users, there are no extensions. If there are users (at least one), they will at least have the 'CRXCavator Gatherer' extension installed. 'CRXCavator Gatherer' is mandatory for user/extension data to get added/updated to the CRX platform.

Reply:
I'm with @sachafaust here.  If there is an error, then `data.raise_for_status()` in `crxcavator.call_crxcavator_api()` will raise an exception and the sync would have failed much earlier on.  I don't see a case for specifically raising an exception if the length of data here is 0.